inc/drivers/MicroBitFile.h

```diff
@@ -0,0 +1,198 @@
+      *         handle is invalid, MICROBIT_INVALID_PARAMETER if bytes is invalid, or
+      *         len is negative.
+      */
+    int write(const char *bytes, int len);
+
+    /**
+      * Writes the given ManagedString to this MicroBitFile instance at the current position.
+      *
+      * @param s The ManagedString to write to this file.
+      *
+      * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+      *         handle is invalid, MICROBIT_INVALID_PARAMETER if bytes is invalid, or
+      *         len is negative.
+      */
+    int write(ManagedString s);
+
+    /**
+      * Reads a single character from the file at the current position.
+      *
+      * @return the character, or MICROBIT_NOT_SUPPORTED if the current file handle
+      *         is invalid.
+      */
+    int read();
+
+    /**
+      * Reads from the file into a given buffer.
+      *
+      * @param buffer a pointer to the buffer where data will be stored.
+      *
+      * @param size the number of bytes that can be safely stored in the buffer.
+      *
+      * @return the number of bytes read, or MICROBIT_INVALID_PARAMETER if buffer is
+      *         invalid, or the size given is less than 0.
+      */
+    int read(char *buffer, int size);
+
+    /**
+      * Reads from the current MicroBitFile into a given buffer.
+      *
+      * @param size the number of bytes to be read from the file.
+      *
+      * @return a ManagedString containing the requested bytes, oran empty ManagedString
+      *         on error.
+      */
+    ManagedString read(int size);
+
+    /**
+      * Removes this MicroBitFile from the MicroBitFileSystem.
+      *
+      * @return MICROBIT_OK on success, MICROBIT_INVALID_PARAMETER if the given filename
+      *         does not exist, MICROBIT_CANCELLED if something went wrong.
+      */
+    int remove();
+
+    /**
+      * Seeks to the end of the file, and appends the given ManagedString to this MicroBitFile instance.
+      *
+      * @param bytes a pointer to the bytes to write to this file.
+      *
+      * @param len the number of bytes to write.
+      *
+      * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+      *         handle is invalid or this file was not opened in WRITE mode.
+      */
+    int append(const char *bytes, int len);
+
+    /**
+      * Seeks to the end of the file, and appends the given ManagedString to this MicroBitFile instance.
+      *
+      * @param s The ManagedString to write to this file.
+      *
+      * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+      *         handle is invalid or this file was not opened in WRITE mode.
+      */
+    int append(ManagedString s);
+
+    /**
+      * Seeks to the end of the file, and appends the given character to this MicroBitFile instance.
+      *
+      * @param c The character to write to this file.
+      */
+    void operator+=(const char c);
```

Comment:
Is this just a wrapper around ``append``? How about not support += since it does not return a proper error code and leads to potentially weird unwanted conversions.